migration.go

```diff
@@ -116,35 +120,76 @@ type Migrations []*Migration
 // Except selects migrations that does not exist in the current ones.
 func (m Migrations) Except(migrations Migrations) (excepted Migrations) {
 	// Mark the current transactions.
-	current := make(map[int64]bool)
+	current := make(map[int64]time.Time)
 	for _, migration := range m {
-		current[migration.Version] = true
+		current[migration.Version] = migration.AppliedAt
 	}
 
 	// Mark the ones in the migrations set, which we do have to get.
-	new := make(map[int64]bool)
+	new := make(map[int64]time.Time)
 	for _, migration := range migrations {
-		new[migration.Version] = true
+		new[migration.Version] = migration.AppliedAt
 	}
 
 	for _, migration := range migrations {
-		if new[migration.Version] && !current[migration.Version] {
+		_, will := new[migration.Version]
+		_, has := current[migration.Version]
+		if will && !has {
 			excepted = append(excepted, migration)
 		}
 	}
 
 	return
 }
 
+// Intersect selects migrations that does exist in the current ones.
+func (m Migrations) Intersect(migrations Migrations) (intersect Migrations) {
```

Comment:
the variables `m`, `migrations` and `intersect` aren't super clear here, can we rename `migrations` to `others` or `intersectWith` and `intersect` with `result`?
Side note, using named returns isn't a good practice unless necessary.